keadm/cmd/keadm/app/cmd/beta/config.go

```diff
@@ -0,0 +1,231 @@
+// TODO: Now we only support docker images. If can, we will need to support more CRIs.
+func newCmdConfigImagesPull() *cobra.Command {
+	cfg := newDefaultConfiguration()
+
+	cmd := &cobra.Command{
+		Use:   "pull",
+		Short: "Pull images used by keadm",
+		RunE: func(_ *cobra.Command, _ []string) error {
+			images := GetKubeEdgeImages(cfg)
+
+			return PullControlPlaneImages(images)
+		},
+		Args: cobra.NoArgs,
+	}
+	AddImagesCommonConfigFlags(cmd, cfg)
+
+	return cmd
+}
+
+// PullControlPlaneImages pulls all images
+func PullControlPlaneImages(images []string) error {
+	for _, image := range images {
+		fmt.Printf("Pulling %s ...\n", image)
+		if err := dockerPullImage(image); err != nil {
+			fmt.Printf("Failed to pull image %q: %v", image, err)
+			return fmt.Errorf("failed to pull image %q: %v", image, err)
+		}
+		fmt.Printf("Successfully pulled %s\n", image)
+	}
+	return nil
+}
+
+// AddImagesCommonConfigFlags adds the flags that configure keadm
+func AddImagesCommonConfigFlags(cmd *cobra.Command, cfg *Configuration) {
+	cmd.Flags().StringVar(&cfg.KubeEdgeVersion, cmdcommon.KubeEdgeVersion, cfg.KubeEdgeVersion,
+		`Use this key to decide which a specific KubeEdge version to be used.`,
+	)
+	cmd.Flags().StringVar(&cfg.ImageRepository, cmdcommon.ImageRepository, cfg.ImageRepository,
+		`Use this key to decide which image repository to pull images from.`,
+	)
+	cmd.Flags().StringVar(&cfg.Part, "part", cfg.Part,
+		"Use this key to set which part keadm will install: cloud part or edge part. If not set, keadm will list/pull all images used by both cloud part and edge part.")
+}
+
+// GetKubeEdgeImages returns a list of container images that related part expects to use
+func GetKubeEdgeImages(cfg *Configuration) []string {
+	var images []string
+
+	part := strings.ToLower(cfg.Part)
+
+	if part == "cloud" {
+		images = append(images, GetKubeEdgeCloudPartImages(cfg)...)
+	} else if part == "edge" {
+		images = append(images, GetKubeEdgeEdgePartImages(cfg)...)
+	} else {
+		// if not specified, will return all images used by both cloud part and edge part
+		images = append(images, GetKubeEdgeCloudPartImages(cfg)...)
+		images = append(images, GetKubeEdgeEdgePartImages(cfg)...)
+	}
+
+	return images
+}
+
+// GetKubeEdgeCloudPartImages returns a list of container images that Cloud part expects to use
+func GetKubeEdgeCloudPartImages(cfg *Configuration) []string {
+	images := []string{}
+
+	// Cloud part images
+	images = append(images, GetImage("admission", cfg))
+	images = append(images, GetImage("cloudcore", cfg))
+	images = append(images, GetImage("iptables-manager", cfg))
+
+	return images
+}
+
+// GetKubeEdgeEdgePartImages returns a list of container images that Edge expects to use
+func GetKubeEdgeEdgePartImages(cfg *Configuration) []string {
+	images := []string{}
+
+	// Edge part images
+	images = append(images, GetPauseImage())
+	images = append(images, GetImage("edgecore", cfg))
```

Comment:
EdgeCore is no needed in edgenode, `installation-package` should be needed in both cloud and edge, cc @zc2638 HDYT?

Reply:
+1

Reply:
fixed